src/macaron/slsa_analyzer/provenance/loader.py

```diff
@@ -103,9 +130,90 @@ def _load_provenance_file_content(
     if not isinstance(json_payload, dict):
         raise LoadIntotoAttestationError("The provenance payload is not a JSON object.")
 
+    if json_payload["predicate"]:
+        return json_payload
+
+    # For provenance without a predicate (e.g. PyPI), try to use the provenance certificate instead.
```

Comment:
I think it would be good to add an if statement to check for predicateType being "https://docs.pypi.org/attestations/publish/v1". 
This is for recognizing if `predicate` being empty is expected (because it's a PYPI attestation) or not (e.g the content if malformed).